oem/ibm/libpldmresponder/file_io.hpp

```diff
@@ -122,13 +122,23 @@ class DMA
     /** @brief Method to fetch the shared memory file descriptor for data
      * transfer
      *
+     * @param[in] isBlckSocket - Varible to decide socket should be blocking /
+     * non blocking mode while opening
+     *
      *  @return returns shared memory file descriptor
      */
-    int getNewXdmaFd()
+    int getNewXdmaFd(bool isBlckSocket = true)
     {
         try
         {
-            xdmaFd = open(xdmaDev, O_RDWR | O_NONBLOCK);
+            if (isBlckSocket)
+            {
+                xdmaFd = open(xdmaDev, O_RDWR | O_NONBLOCK);
```

Comment:
here you are checking if its a blockSocket and then opening it in O_NONBLOCK mode.

Reply:
yea correct, will change